text/0000-nested-publish.md

```diff
@@ -0,0 +1,359 @@
+- Feature Name: `nested_publish`
+- Start Date: 2023-06-30
+- RFC PR: [rust-lang/rfcs#3452](https://github.com/rust-lang/rfcs/pull/3452)
+- Rust Issue: ...
+
+# Summary
+[summary]: #summary
+
+Allow published Cargo packages to depend on other packages stored within themselves, as is currently possible in unpublished packages.
+
+# Motivation
+[motivation]: #motivation
+
+There are a number of reasons why a Rust developer currently may feel the need to create multiple library crates, and therefore multiple Cargo packages (since one package contains at most one library crate). These multiple libraries could be:
+
+* A trait declaration and a corresponding derive macro (which must be defined in a separate proc-macro library).
+* A library that uses a build script that uses another library or binary (e.g. for precomputation or bindings generation).
+* A logically singular library broken into multiple parts to speed up compilation.
```

Comment:
> A logically singular library broken into multiple parts to speed up compilation.

Similarly, a `bin` might want to split out a `lib` for local development and testing but not consider it public and not offer semver guarantees for the `lib`.  `cargo-edit` and `cargo-release` are like this.